### ISSUE TITLE
Error mjml.mjml2html is not a function in  Subscription Form

### ISSUE THREAD

**Issue by maxsarlija** [Mailtrain-org/mailtrain#569]
Hello to want to access the subscription form of the list gives error mjml.mjml2html is not a function
, is there any html form available for automatic subscription to the lists?

**Comment by bures** [comment]
Are you using Mailtrain v1 or v2?

**Comment by maxsarlija** [comment]
Hi Bures, Im usign this version: https://github.com/Mailtrain-org/mailtrain/archive/master.zip i follow this tutorial: https://www.scaleway.com/docs/install-mailtrain-mailing-tool/

**Comment by maxsarlija** [comment]
> Are you using Mailtrain v1 or v2?

Hi Bures, Im usign this version: https://github.com/Mailtrain-org/mailtrain/archive/master.zip i follow this tutorial: https://www.scaleway.com/docs/install-mailtrain-mailing-tool/

**Comment by bures** [comment]
I see. It's v1. I'm not sure if I can help you with that. However, if you feel a bit adventurous, you can try our v2 beta. It come with install script for Ubuntu 18.04 and Centos 7. https://github.com/Mailtrain-org/mailtrain/tree/development

**Comment by maxsarlija** [comment]
> I see. It's v1. I'm not sure if I can help you with that. However, if you feel a bit adventurous, you can try our v2 beta. It come with install script for Ubuntu 18.04 and Centos 7. https://github.com/Mailtrain-org/mailtrain/tree/development

Try installing the beta in Centos and Ubuntu but at the end and want to start the service it says: unit not found systemctl

**Comment by bures** [comment]
Can you send the full output of the install script and tell me which OS you use and how did you install it? E.g. whether it was manual install of the minimal profile, or AWS image, or ...

**Comment by maxsarlija** [comment]
> Can you send the full output of the install script and tell me which OS you use and how did you install it? E.g. whether it was manual install of the minimal profile, or AWS image, or ...

To virtualize I use proxmox and the default images of CENTOS 7 and Ubuntu 18.04 Minimal, the process is completed without errors but when wanting to run the mailtrain service is not recognized and gives the same error, do you have any suggestions?
Another query, do you have an html form for subcripsion directly impacting the API?

**Comment by bures** [comment]
How exactly do you start the service?

**Comment by bures** [comment]
As for the question about HTML form, I'm not sure I understand.

**Comment by maxsarlija** [comment]
> How exactly do you start the service?

I follow the steps of the tutorial: 
5- Start Mailtrain and enable to be started by default when your server starts.

systemctl start mailtrain
systemctl enable mailtrain
it says: unit not found systemctl

**Comment by maxsarlija** [comment]
> As for the question about HTML form, I'm not sure I understand.

I need to integrate in my page a subcription form with html code

**Comment by bures** [comment]
What do you get when you try `systemctl list-unit-files`?

As for the HTML form. You can create the form yourself and submit it an a similar way the mailtrain forms are submitted. We don't have any specific support for this. There used to be an embeddable widget in Mailtrain v1, but we have not yet ported this for v2.

**Comment by maxsarlija** [comment]
> What do you get when you try `systemctl list-unit-files`?
> 
> As for the HTML form. You can create the form yourself and submit it an a similar way the mailtrain forms are submitted. We don't have any specific support for this. There used to be an embeddable widget in Mailtrain v1, but we have not yet ported this for v2.

Well, do you have an example of a form with post to the api of mailtrain to send a new subscriber?
Something similar this: https://ibb.co/9W04zqz

I'm going to install V2 again on a test server and send you the log

Thanks in advance.

**Comment by linuxwhy** [comment]
Clicking unsubscription link leads to the same error. I guess some upgrade in node modules breaks the V1, because I've used the V1 for a while, I only encountered this problem recently. Please someone takes a look. Every new user will get the error I suppose, and having a very bad first impression

**Comment by maxsarlija** [comment]
> Clicking unsubscription link leads to the same error. I guess some upgrade in node modules breaks the V1, because I've used the V1 for a while, I only encountered this problem recently. Please someone takes a look. Every new user will get the error I suppose, and having a very bad first impression

Yes, i have a lots of problems in V1

**Comment by linuxwhy** [comment]
> > Clicking unsubscription link leads to the same error. I guess some upgrade in node modules breaks the V1, because I've used the V1 for a while, I only encountered this problem recently. Please someone takes a look. Every new user will get the error I suppose, and having a very bad first impression
> 
> Yes, i have a lots of problems in V1

I only met this problem, and I solved it by changing several lines of code.  From mjml3 to mjml4, mjml2html is now the default export, so google and change  files 
sudo vi lib/helpers.js
sudo vi lib/mailer.js
sudo vi lib/models/forms.js

**Comment by bures** [comment]
Perfect. please submit a PR. I'll be happy to accept it.

**Comment by linuxwhy** [comment]
Now I find some other new problems, like bounced and complains are always 0. (It's not right, cause if I check the statistics on amazon, there are some bounced and complaints), so I have to dig in, and maybe move to the V2 asap.

**Comment by linuxwhy** [comment]
I moved server recently, and I don't know somehow the 25 port was blocked by ufw. I messed up. Problems were caused mainly by this server replacing.

**Comment by maxsarlija** [comment]
I still have the same problem when I want to use the subcription form, do you have an alternative or some html form that solves this?

![image](https://user-images.githubusercontent.com/29083745/59036817-6845bd80-8846-11e9-9262-88707f0abd7e.png)

**Comment by maxsarlija** [comment]
Error in: http://nx08.republicaservers.com:3000/subscription/mBZLF0L7j

**Comment by forgeOneHQ** [comment]
@maxsarlija please try the latest version of v2 and let us know if you still run into issues. You are welcome to provide all infos in a new issue if you run into issues.
